src/details/ArborX_DetailsCrsGraphWrapperImpl.hpp

```diff
@@ -150,9 +146,8 @@ void queryImpl(ExecutionSpace const &space, Tree const &tree,
   {
     tree.query(
         space, permuted_predicates,
-        InsertGenerator<FirstPassTag, PermutedPredicates, Callback, OutputView,
-                        CountView, PermutedOffset>{callback, out, counts,
-                                                   permuted_offset},
+        InsertGenerator<FirstPassTag, Callback, OutputView, CountView,
```

Comment:
Do you need to specify the template parameters other than the tag?

Reply:
I tried removing them, and it failed. I always forget what the rules are.

Reply:
If one template parameter is specified, all of them need to be provided, see https://godbolt.org/z/eKhnTW3nr. Aggregate deduction only works if no template parameters are given (which would require storing the tag here).

Reply:
Thank you.